### ISSUE TITLE
skip-covered with 100% coverage throws a "no data" error

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#549]
*Originally reported by* **Havoc Pennington (Bitbucket: [havocp](https://bitbucket.org/havocp), GitHub: [havocp](https://github.com/havocp))**

----------------------------------------

We run tests something like this:

```
#!shell

python -m pytest -vv --cov=modname --cov-report=term-missing:skip-covered --cov-report=html --cov-fail-under=100 --no-cov-on-fail --cov-config=.coveragerc modname/tests
```

With 4.2, with 100% coverage this exited successfully, with 4.3.1 it results in "ERROR: Failed to generate report: No data to report."

If I add uncovered code (and remove the fail-under=100), then it exits successfully. If I remove skip-covered then it exits successfully.

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/549

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

**skip-covered** was not implemented in 4.2 and was silently ignored. It was implemented in 4.3 and it looks like the 100% coverage case was forgotten.

**Comment by nedbat** [comment]
*Original comment by* **Havoc Pennington (Bitbucket: [havocp](https://bitbucket.org/havocp), GitHub: [havocp](https://github.com/havocp))**

----------------------------------------

skip-covered worked for me in 4.2 for the term-missing report but not the html report.

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

The **skip-covered** option is global. If set once it will apply to all reports that support it. This is why it applies to the html report although it is set for term-missing only.

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

For the record https://bitbucket.org/ned/coveragepy/pull-requests/123/html-total-must-account-for-files-100/diff is the proposed fix for this issue

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

@nedbat is there a way to tag this as a regression introduced in 4.3 ?

**Comment by nedbat** [comment]
html total must account for files 100% covered #549

The --skip-covered implementation for html must not just skip the
files that are 100% covered. It must remember them and not fail with a
"no coverage data" when all files are 100% covered.

The statistics about the files that are 100% covered must also be
summed in the total for the index.html page.

close #549

→ <<cset [9158591bba46 (bb)](https://bitbucket.org/ned/coveragepy/commits/9158591bba46)>>

**Comment by nedbat** [comment]
*Original comment by* **Havoc Pennington (Bitbucket: [havocp](https://bitbucket.org/havocp), GitHub: [havocp](https://github.com/havocp))**

----------------------------------------

Thanks!

**Comment by nedbat** [comment]
This was shipped in 4.3.2.
